pac4j-core/src/main/java/org/pac4j/core/profile/InternalAttributeHandler.java

```diff
@@ -0,0 +1,128 @@
+package org.pac4j.core.profile;
+
+import org.pac4j.core.profile.converter.Converters;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.JavaSerializationHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Internally handles attributes (set / get).
+ *
+ * @author Jerome Leleu
+ * @since 2.0.0
+ */
+public class InternalAttributeHandler implements Serializable {
```

Comment:
I may be late to this party but if the idea is that this should not be treated as an external API, it might be useful for the documentation to specify that this is not part of the public API and can be removed or changed without notice, so should not generally be used by code external to pac4j (assuming we can't tweak the visibility to something other than public).